drake/core/Function.h

```diff
@@ -104,46 +106,54 @@ namespace Drake {
     }
   };
 
-  class Function {
-  public:
-    Function() : relation(InputOutputRelation::Form::ARBITRARY) {};
-    Function(InputOutputRelation::Form f) : relation(f) {};
-    virtual ~Function() {};
+  template <typename ScalarType> using VecIn = Eigen::Ref< Eigen::Matrix<ScalarType, Eigen::Dynamic, 1> const >;
+  template <typename ScalarType> using VecOut = Eigen::Matrix<ScalarType, Eigen::Dynamic, 1>;
```

Comment:
Agreed that we don’t need to block on this. But my sense is that we will want to
have a minimal number of vector typedefs kicking around, and to accumulate them
in Vector.h. I thought we already had some for these, but see now that you’re
correct (we don’t).